client/src/components/container/user/EventPage/index.js

```diff
@@ -1,27 +1,50 @@
 import React, { Component } from 'react'
-import { Button } from "antd";
+import { Button ,Alert} from "antd";
 
 import EventPageContent from '../EventPageContent/EventPageContent'
 import PopupBtnEventcomp from '../PopupBtnEventcomp/PopupBtnEventcomp'
+import Axios from 'axios';
 
 import './style.css'
 
 class EventPage extends Component {
-
-  render() {
+  state={
+    image:"",
+    title:"",
+    otherEventProps:[],
+    error:""
+  }
+  componentDidMount(){
     const { location: { state: {info : { image, ...otherEventProps }} } } = this.props;
     const { location: { state: {info:{title}} } } = this.props;
+    const { match: { params: { eventCode } } } = this.props;
+    if(this.props.location.state.info){
+      this.setState({image,title,otherEventProps})
+    }else{
+      Axios.get(`/api/v1/event/${eventCode}`)
+      .then(({data})=>{
+        const {image,title,...otherEventProps}=data
+        this.setState({image,title,otherEventProps})
+      })
+      .catch(error=>{
+        this.setState({error})
+      })
+  }
+}
+
+  render() {
     const { history: { goBack, push } } = this.props;
     const { match: { params: { eventCode, eventProg } } } = this.props;
-
+    const {image,title,otherEventProps,error}=this.state
     return (
       <div className='container'>
-        <img src={image} alt='event page background' className='image' />
+        {(error)&&(<Alert type="error">cant find event</Alert>)}
+        <img src={(image)?image:this.state.image} alt='event page background' className='image' />
```

Comment:
why did you make a condition here ?! every time you will take the image from state.
So fix it here : 
```suggestion
        <img src={image} alt='event page background' className='image' />
```